core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.controller.ts

```diff
@@ -100,6 +146,94 @@ angular.module('oppia').controller('TranslationModalController', [
       $scope.activeWrittenTranslation.html = '';
     };
 
+    $scope.getElementTexts = function(elements, type) {
+      const textWrapperLength = 6;
+      const attributes = Array.from(elements, function(element: HTMLElement) {
+        if (element.localName === 'oppia-noninteractive-image') {
+          const attribute = element.attributes[type].value;
+          return attribute.substring(
+            textWrapperLength, attribute.length - textWrapperLength);
+        }
+      });
+      return attributes.filter(attribute => attribute);
+    };
+
+    $scope.copiedAllElements = function(
+        originalElements, translatedElements) {
+      const hasMatchingTranslatedElement = (element) => (
+        translatedElements.includes(element));
+      return originalElements.every(hasMatchingTranslatedElement);
+    };
+
+    $scope.hasSomeDuplicateElements = function(
+        originalElements, translatedElements) {
+      if (originalElements.length === 0) {
+        return false;
+      }
+      const hasMatchingTranslatedElement = (element) => (
+        translatedElements.includes(element) && originalElements.length > 0);
+      return originalElements.some(hasMatchingTranslatedElement);
+    };
+
+    $scope.getTexts = function(rawText) {
+      const foundImageFilePaths = $scope.getElementTexts(
+        rawText, 'filepath-with-value');
+      const foundImageAlts = $scope.getElementTexts(
+        rawText, 'alt-with-value');
+      const foundImageDescriptions = $scope.getElementTexts(
+        rawText, 'caption-with-value');
+
+      return {
+        foundImageFilePaths,
+        foundImageAlts,
+        foundImageDescriptions
+      };
+    };
+
+    $scope.isTranslationCompleted = function(
+        originalElements: [HTMLElement], translatedElements: [HTMLElement]) {
+      const originalHtmlElements = Array.from(
```

Comment:
`HtmlElements` is a bit misleading, this should probably be something `originalElementTagNames`. In general, make sure your variable names are precise and descriptive.